src/backend/TrafficCourts/Citizen.Service/Features/Tickets/Search.cs

```diff
@@ -1,179 +1,126 @@
+        /// <param name="time"></param>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <exception cref="ArgumentException"></exception>
+        public Request(string ticketNumber, string time)
         {
-            private Response()
-            {
-            }
+            ArgumentNullException.ThrowIfNull(ticketNumber);
+            ArgumentNullException.ThrowIfNull(time);
 
-            public Response(SearchReply reply)
+            if (!Regex.IsMatch(ticketNumber, TicketNumberRegex))
             {
-                ArgumentNullException.ThrowIfNull(reply);
-                if (reply.ViolationTicketNumber is null) throw new ArgumentException("Search reply does not contain a violation ticket", nameof(reply));
-                if (reply.ViolationDate is null) throw new ArgumentException("Search reply does not contain a violation date", nameof(reply));
-                if (reply.ViolationTime is null) throw new ArgumentException("Search reply does not contain a violation time", nameof(reply));
-
-                TicketSearchResult ticketSearchResult = new()
-                {
-                    ViolationTicketNumber = reply.ViolationTicketNumber,
-                    ViolationDate = new DateTime(reply.ViolationDate.Year, reply.ViolationDate.Month, reply.ViolationDate.Day),
-                    ViolationTime = $"{reply.ViolationTime.Hour:d2}:{reply.ViolationTime.Minute:d2}"
-                };
-
-                foreach (Offence offence in reply.Offences)
-                {
-                    TicketOffence ticketOffence = new()
-                    {
-                        AmountDue = offence.AmountDue / 100m,
-                        OffenceDescription = offence.OffenceDescription,
-                        VehicleDescription = offence.VehicleDescription,
-                        OffenceNumber = offence.OffenceNumber,
-                        TicketedAmount = offence.TicketedAmount / 100m,
-                        InvoiceType = offence.InvoiceType
-                    };
-
-                    ticketSearchResult.Offences.Add(ticketOffence);
-                }
-
-                Result = ticketSearchResult;
+                throw new ArgumentException("ticketNumber must start with two upper case letters and 6 or more numbers", nameof(ticketNumber));
             }
 
-            public Response(Exception exception)
+            if (!Regex.IsMatch(time, TimeRegex))
             {
-                ArgumentNullException.ThrowIfNull(exception);
-                Result = exception;
+                throw new ArgumentException("time must be properly formatted 24 hour clock", nameof(time));
             }
 
-            /// <summary>
-            /// The result value.
-            /// </summary>
-            public OneOf<TicketSearchResult, Exception> Result { get; }
+            TicketNumber = ticketNumber;
+            Hour = int.Parse(time[0..2]);
+            Minute = int.Parse(time[3..5]);
```

Comment:
Are there xUnit tests for this time indexing logic?   From the regex, it looks like the minute portion could be 1 digit instead of 2, in which case this may throw an index out of bounds exception.